include/SFML/System/Clock.hpp

```diff
@@ -74,11 +75,23 @@ class SFML_SYSTEM_API Clock
     Time restart();
 
 private:
+    using ClockImpl = std::chrono::system_clock;
```

Comment:
`sf::Clock` is not a date/time API, but a monotonic high-precision clock. As such, `system_clock` is the wrong choice.

From https://en.cppreference.com/w/cpp/chrono/system_clock:

> Class std::chrono::system_clock represents the system-wide real time wall clock.
> 
> It may not be monotonic: on most systems, the system time can be adjusted at any moment. It is the only C++ clock that has the ability to map its time points to C-style time. 

We should probably use [`high_resolution_clock`](https://en.cppreference.com/w/cpp/chrono/high_resolution_clock) instead. It could be that they end up using the same implementation, but this is the platform's choice.

Reply:
A good thing to note about `high_resolution_clock` which leans me more towards `steady_clock`:

>The high_resolution_clock is not implemented consistently across different standard library implementations, and its use should be avoided. It is often just an alias for std::chrono::steady_clock or std::chrono::system_clock, but which one it is depends on the library or configuration. When it is a system_clock, it is not monotonic (e.g., the time can go backwards). For example, for gcc's libstdc++ it is system_clock, for MSVC it is steady_clock, and for clang's libc++ it depends on configuration.

>Generally one should just use std::chrono::steady_clock or std::chrono::system_clock directly instead of std::chrono::high_resolution_clock: use steady_clock for duration measurements, and system_clock for wall-clock time.

Specifically:
> use steady_clock for duration measurements

That seems to be what sf::Clock needs to represent. So how about `std::chrono::steady_clock`?

Reply:
I'll answer in the main thread, since it concerns the PR as a whole.